src/wobjectdefs.h

```diff
@@ -199,41 +215,213 @@ constexpr int summed = sums(Args...);
-template<std::size_t N, std::size_t... I> struct StaticString<N, std::index_sequence<I...>>
-{
-    StaticStringArray<N> data;
-    constexpr StaticString(StaticStringArray<N> &d) : data{ (d[I])... } { }
+template<std::size_t N> struct StaticString {
+    char data[N] = {};
     static constexpr std::size_t size = N;
-    constexpr char operator[](int p) const { return data[p]; }
+
+    constexpr StaticString() = default;
+    template<size_t L>
+    constexpr StaticString(const char (&d)[L]) {
+        static_assert (L == N);
+        auto p = data;
+        for (auto c : d) *p++ = c;
+    }
+};
+template<size_t N> StaticString(const char (&)[N]) -> StaticString<N>;
+
+template<size_t... Vs> struct IndexPack {};
+template<size_t... Vs> constexpr auto index_pack = IndexPack<Vs...>{};
+
+template<class T, size_t N> using RawArray = T[N];
+
+template<size_t N>
+struct IndexArray {
+    size_t data[N] = {};
+    static constexpr size_t size = N;
+
+    constexpr IndexArray() = default;
+    template<size_t... Vs> constexpr IndexArray(IndexPack<Vs...>) : data{Vs...} {
+        static_assert (N == sizeof... (Vs));
+    }
+
+    constexpr auto operator[](size_t i) const noexcept -> size_t { return data[i]; }
 };
-template <std::size_t N> constexpr StaticString<N> makeStaticString(StaticStringArray<N> &d) { return {d}; }
-
-/** A list containing many StaticString with possibly different sizes */
-template<typename T = void> using StaticStringList = binary::tree<T>;
-
-/** Make a StaticStringList out of many char array  */
-constexpr StaticStringList<> makeStaticStringList() { return {}; }
-template<typename... T>
-constexpr StaticStringList<> makeStaticStringList(StaticStringArray<1> &, T...)
-{ return {}; }
-template<std::size_t N, typename... T>
-constexpr auto makeStaticStringList(StaticStringArray<N> &h, T&...t)
-{ return binary::tree_prepend(makeStaticStringList(t...), StaticString<N>(h)); }
-template<typename... T>
-constexpr StaticStringList<> makeStaticStringList(StaticString<1>, T...)
-{ return {}; }
-template<std::size_t N, typename... T>
-constexpr auto makeStaticStringList(StaticString<N> h, T...t)
-{ return binary::tree_prepend(makeStaticStringList(t...), h); }
-
-
-/** Add a string in a StaticStringList */
-template<std::size_t L, typename T>
-constexpr auto addString(const StaticStringList<T> &l, const StaticString<L> & s) {
-    return binary::tree_append(l, s);
+template<size_t... Vs> IndexArray(IndexPack<Vs...>) -> IndexArray<sizeof...(Vs)>;
+
+
+template<size_t... Sizes> constexpr auto toOffsetArray() {
+    auto offsets = IndexArray<sizeof...(Sizes)>{};
+    auto o = size_t{};
+    auto p = offsets.data;
+    ((*p++ = o, o += Sizes),...);
+    return offsets;
+}
+template<size_t I, size_t... Sizes> constexpr auto getOffset() -> size_t {
+    auto i = 0u;
+    auto o = size_t{};
+    ((i++ < I ? o += Sizes : 0u), ...);
+    return o;
 }
 
+template<size_t... Ns>
+struct StaticStrings {
```

Comment:
Naming is hard… I needed a name that was different, because both ran in parallel for a while.
* `StaticStringArray` was already taken.
* `StaticStringList` feels ab bit wrong, because there is not list involved.
* I also agree that they are visually very similar and easy to confuse. Surprisingly this never happened to me so far.

I have no affinity these names, consider them as a first suggestion to get the code running. I will use whatever name is considered best.